### ISSUE TITLE
Block toolbar: Buttons that have visible text should not use tooltips

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#42811]
### Description

Similar to https://github.com/WordPress/gutenberg/issues/42676

When the preference 'Show button text labels' is enabled, the Block toolbar buttons show text instead of icons. However, these buttons still use a tooltip that appears on hover and focus. The tooltip just repeats the visible button text and it just clutters the UI.

<img width="624" alt="Screenshot 2022-07-29 at 11 41 09" src="https://user-images.githubusercontent.com/1682452/181733086-7b59c59f-583d-4f0e-baaa-615762cdf17e.png">

In the Top toolbar there's a convention in place when 'Show button text labels' is enabled:
- Only buttons that have an associated keyboard shortcut still display a tooltip (to visually expose the shortcut)
- All the other buttons don't show a tooltip.

For consistency and better UI, this convention should be used also on the Block toolbar.

### Step-by-step reproduction instructions

- Go to Options > Preferences, and enable 'Show button text labels'.
- Select any block to make its toolbar appear.
- Hover or focus the toolbar buttons.
- Observe they all show a tooltip.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
Note: the 'Move up' and 'Move down' buttons need more horizontal space. I think there's already an issue for that.
